### ISSUE TITLE
[MU4 Task] Implement ‘Save diagnostic files‘ flow

### ISSUE THREAD

**Issue by bkunda** [musescore/MuseScore#15830]
**Task description**
This is a new feature that allows the user to quickly save relevant files as a .zip file so they can send it to developers when asking for technical support. 

The design consists of:

1. New menu item in **Diagnostic** --> **Save diagnostic files**
2. New dialog (triggered by the above menu option):

<img width="527" alt="ExportApplicationSupportFiles-1" src="https://user-images.githubusercontent.com/86290556/212032077-de39bd78-6ea6-47e1-bae6-8ddef5c2df5b.png">

_N.B. The ‘Open destination folder after saving‘ checkbox would be **unchecked** by default_

3. Clicking **Save** triggers the user‘s standard operating system ‘Save‘ dialog.

The files being exported will be drawn from `Application Support` and will include: 

1.  shortcuts.xml/midi_mappings.xml
2. information about found VST (see the vst folder)
3. workspaces
4. logs/dumps
5. active qml plugins  

Any *.mscz files saved in this location (e.g. recovery files) will be ignored in this export.

Thanks!

@shoogle @Tantacrul @vpereverzev 
@RomanPudashkin feel free to re-assign to another project/label as desired :-)

**Comment by DmitryArefiev** [comment]
Implemented in #15907 (with some design changes comparing with the description)

**Comment by RomanPudashkin** [comment]
@bkunda FYI 

Design changes:

1. The "Open destination folder" checkbox has been replaced by "Don't show again". It's more convenient for our testers because they will open this dialog very often (so they can hide it and save some time)
2. We will always open the destination folder after saving diagnostic files

The main points why I decided to remove the "Open destination folder" checkbox from the dialog are as follows:
* to allow us to reuse our standard dialog component (we just pass it our text and that's it), which already supports navigation/accessibility
* we don't need to create an entirely new dialog component just for one checkbox or modify the standard component to introduce more instability

The dialog can still be improved in the future
